crates/matrix-sdk-indexeddb/src/state_store/migrations.rs

```diff
@@ -378,14 +408,21 @@ async fn migrate_to_v3(db: &IdbDatabase, store_cipher: Option<&StoreCipher>) ->
     v3_fix_store(&tx.object_store(keys::ROOM_STATE)?, store_cipher).await?;
     v3_fix_store(&tx.object_store(keys::ROOM_INFOS)?, store_cipher).await?;
 
-    tx.await.into_result().map_err(|e| e.into())
+    tx.await.into_result()?;
+
+    db.close();
+
+    // Update the version of the database.
+    let db = IdbDatabase::open_u32(name, 3)?.into_future().await?;
```

Comment:
Here (and a few times below), we're closing a database we're just about to reopen one statement later. Could we avoid that, one way or another?